src/Hostname.php

```diff
@@ -1772,8 +1772,16 @@ public function isValid($value)
                     // id-prefix: alpha / digit
                     // ldh: alpha / digit / dash
 
-                    // Match TLD against known list
                     $this->tld = strtoupper($matches[1]);
+                    // Decode Punycode TLD to IDN
+                    if (strpos($this->tld, 'xn--') === 0) {
+                        $this->tld = $this->decodePunycode(substr($this->tld, 4));
```

Comment:
Setting multibyte values in the property breaks non mutlibyte methods `if (!in_array(strtolower($this->tld), $this->validTlds)`